File: server/utils/flags/feature_flags.py
"""Feature flags for toggling functionality.
Uses NEXT_PUBLIC_ENABLE_* variables shared with frontend for single source of truth.
"""
import os

def is_ovh_enabled() -> bool:
    """Check if OVH integration is enabled via environment variable."""
    return os.getenv("NEXT_PUBLIC_ENABLE_OVH", "false").lower() == "true"


def is_pagerduty_oauth_enabled() -> bool:
    """Check if PagerDuty OAuth integration is enabled via environment variable."""
    return os.getenv("NEXT_PUBLIC_ENABLE_PAGERDUTY_OAUTH", "false").lower() == "true"


def is_confluence_enabled() -> bool:
    """Confluence is always enabled (no longer feature-gated)."""
    return True


def is_jira_enabled() -> bool:
    """Check if Jira integration is enabled via environment variable."""
    return os.getenv("NEXT_PUBLIC_ENABLE_JIRA", "false").lower() == "true"


def is_sharepoint_enabled() -> bool:
    """Check if SharePoint integration is enabled via environment variable."""
    return os.getenv("NEXT_PUBLIC_ENABLE_SHAREPOINT", "false").lower() == "true"


def is_spinnaker_enabled() -> bool:
    """Check if Spinnaker integration is enabled via environment variable."""
    return os.getenv("NEXT_PUBLIC_ENABLE_SPINNAKER", "false").lower() == "true"


def is_incident_prevention_enabled() -> bool:
    """Check if PR change-gating (Incident Prevention) is enabled."""
    return os.getenv("NEXT_PUBLIC_ENABLE_INCIDENT_PREVENTION", "true").lower() == "true"

File: client/src/lib/feature-flags.ts
/**
 * Feature flags for toggling functionality.
 * Uses NEXT_PUBLIC_ENABLE_* variables shared with backend for single source of truth.
 */

import { getEnv } from '@/lib/env';

export const isPagerDutyOAuthEnabled = () => {
  return getEnv('NEXT_PUBLIC_ENABLE_PAGERDUTY_OAUTH') === 'true';
};

export const isOvhEnabled = () => {
  return getEnv('NEXT_PUBLIC_ENABLE_OVH') === 'true';
};

export const isScalewayEnabled = () => {
  return getEnv('NEXT_PUBLIC_ENABLE_SCALEWAY') === 'true';
};

export const isSharePointEnabled = () => {
  return getEnv('NEXT_PUBLIC_ENABLE_SHAREPOINT') === 'true';
};

export const isConfluenceEnabled = () => {
  return getEnv('NEXT_PUBLIC_ENABLE_CONFLUENCE') === 'true';
};

export const isJiraEnabled = () => {
  return getEnv('NEXT_PUBLIC_ENABLE_JIRA') === 'true';
};

export const isSpinnakerEnabled = () => {
  return getEnv('NEXT_PUBLIC_ENABLE_SPINNAKER') === 'true';
};
